### PR TITLE
Ruby on Rails capstone project

### PR DESCRIPTION
**Pull request by LionRouge1**
Interactions

    Splash screen
        A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

    Sign up and log in pages
        The user should be able to register in the app with full name, email and password (all mandatory).
        The user can log into the app using email and password.
        If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

    Home page (catagories page)
        When the user logs in, they are presented with the categories page.
        For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
        When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
        There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

    Transactions page
        For a given category, the list of transactions is presented, ordered by the most recent.
        At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
        There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
        When the user clicks on the "Back" button (<), the user navigates to the home page.

    "Add a new category" page
        The user fills out a form to create a new category, indicating their name and icon (both mandatory).
        The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
        When the user clicks on the "Back" button (<), the user navigates to the home page.

    "Add a new transaction" page
        The user fills out a form to create a new transaction with:
            name (mandatory)
            amount (mandatory)
            categories (mandatory at least one)
        The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
        When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

Testing requirements

    Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
```
Comment (KDlamini) [inline README.md]: Kindly make sure that the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        <%= f.check_box :remember_me %>
+        <%= f.label :remember_me %>
+      </div>
+    <% end %>
+  </div>
+
+  <div class="links">
+    <%= f.submit "Log in", class: "btn btn-sign_in" %>
+    <%= link_to "Forgot your password?", new_password_path(resource_name), class: "btn btn-sign_up" %>
+  </div>
```
Comment (KDlamini) [inline app/views/devise/sessions/new.html.erb:17]: I am able to sign up and I am automatically logged in after registration. Great job on implementing user registration properly.

- However, if I log out and try to log in again nothing happens. log in is currently not working. Kindly look into it.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @LionRouge1 🙋‍♀️,  

Great work so far! 💯 
You have done an incredible job creating a working budget app according to the project requirements. 🎊 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  App follows the design and entity relationship diagram
✔️  All unit and integration tests are written
✔️  Correct PR title and description
✔️  Correct use of Gitflow 
✔️  No linter errors

## Required Changes ♻️

**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (LionRouge1) [reply app/views/devise/sessions/new.html.erb:17]: Hi @KDlamini,
Thanks for the review, I just tried to sign up and log out with a new credential, and log in with the new user, and everything is working fine :smiley_cat: . Please try to pull from the `dev` branch and see. You can also try the online version [here](https://budgetapplion.herokuapp.com/).
Thanks

**Review by descholar-ceo — APPROVED** [review]
Hi @LionRouge1 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Your project looks fine ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**